libctru/include/3ds/svc.h

```diff
@@ -73,13 +73,16 @@ typedef struct {
 } PageInfo;
 
 typedef enum {
-	ARBITER_FREE            = 0,
-	ARBITER_ACQUIRE         = 1,
-	ARBITER_KERNEL2         = 2,
-	ARBITER_ACQUIRE_TIMEOUT = 3,
-	ARBITER_KERNEL4         = 4,
+	ARBITRATION_SIGNAL                                  = 0, ///< Signal #value threads for wake-up.
+	ARBITRATION_WAIT_IF_LESS_THAN                       = 1, ///< If the memory at the address is strictly lower than #value, then wait for signal.
+	ARBITRATION_DECREMENT_AND_WAIT_IF_LESS_THAN         = 2, ///< If the memory at the address is strictly lower than #value, then decrement it and wait for signal.
```

Comment:
Did you check what was then the value of the address while the thread waited, and then afterwards after it was signalled?